File: Golang/Check Completeness of a Binary Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isCompleteTree(root *TreeNode) bool {
    if root == nil {
        return true
    }

    q := []*TreeNode{root}
    flag := false
    for len(q) > 0 {
        node := q[0]
        q = q[1:]
        if node == nil {
            flag = true
        } else {
            if flag {
                return false
            }
            q = append(q, node.Left, node.Right)
        }
    }
    return true
    
}

File: Python3/Check Completeness of a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isCompleteTree(self, root: Optional[TreeNode]) -> bool:
        if not root: return True
        q = [root]
        flag = False

        while q:
            node = q.pop(0)
            if node is None:
                flag = True
                continue

            if flag:
                return False

            q.append(node.left)
            q.append(node.right)
        
        return True